pkg/sources/source.go

```diff
@@ -136,6 +136,17 @@ func (sp *SourceProcessor) getSourcer(writers []isb.BufferWriter, fetchWM fetch.
 	src := sp.VertexInstance.Vertex.Spec.Source
 
 	if x := src.Generator; x != nil {
+		// generator fields can be nil if using minimal CRD that doesn't provide defaults
+		if x.RPU == nil {
+			return nil, fmt.Errorf("generator rpu is not set")
```

Comment:
instead of throwing an error we could set here to the defaults, but then we are duplicating defaults present in the kubebuilder annotations (which populate the full crd).  but seems like we are already doing that for replicas. ref: 

* https://github.com/numaproj/numaflow/blob/27e6a8755bd1092b31d93fac891d1decd006a093/pkg/apis/numaflow/v1alpha1/redis_buffer_service.go#L124
* https://github.com/numaproj/numaflow/blob/27e6a8755bd1092b31d93fac891d1decd006a093/pkg/apis/numaflow/v1alpha1/redis_buffer_service.go#L44

maybe best not use crd defaults at all & just default in code?

anyway, ya'll let me know what changes are desired.

Reply:
Add the validation to https://github.com/numaproj/numaflow/blob/main/pkg/reconciler/pipeline/validate.go